### PR TITLE
helm chart update

### PR DESCRIPTION
**Pull request by Nagarajan0396**

### CODE REVIEW

File: deployments/discovery-engine-chart/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v2
+appVersion: 1.16.0
+description: A Helm chart for DiscoveryEngine
```
Comment (nyrahul) [inline deployments/discovery-engine-chart/Chart.yaml]:
```suggestion
description: A Helm chart for AccuKnox DiscoveryEngine

```

File: deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+application:
+  name: knoxautopolicy
+  network:
+    operation-mode: 1                         # 1: cronjob | 2: one-time-job
+    cron-job-time-interval: "0h0m10s"         # format: XhYmZs
+    operation-trigger: 1000
+    network-log-from: "hubble"                # db|hubble|feed-consumer
```
Comment (nyrahul) [inline deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml]: This configuration is wrong. Please check the latest config in src/conf/local-file.yaml

**Review by nyrahul — APPROVED** [review]
Two minor comments. Please check and handle, otherwise all good.

**Comment by nyrahul** [comment]
Hey @Nagarajan0396 , thank you for working on this. Lets handle few basic comments above and this should be good to merge.

**Comment by Nagarajan0396** [comment]
Hi @nyrahul  Changes has been done.

**Review by nyrahul — APPROVED** [review]

File: deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml [hunk 3]
```diff
@@ -0,0 +1,103 @@
+application:
+  name: knoxautopolicy
+  network:
+    operation-mode: 1                             # 1: cronjob | 2: one-time-job
+    operation-trigger: 5
+    cron-job-time-interval: "0h0m10s"             # format: XhYmZs 
+    network-log-limit: 10000
+    network-log-from: "kubearmor"                 # db|hubble|feed-consumer|kubearmor
+    #network-log-file: "/home/rahul/feeds.json"   # file path
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml]:
```suggestion
```

File: deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    namespace-filter:
+      - "!kube-system"
+  system:
+    operation-mode: 1                         # 1: cronjob | 2: one-time-job
+    operation-trigger: 5
+    cron-job-time-interval: "0h0m10s"         # format: XhYmZs
+    system-log-from: "kubearmor"              # db|kubearmor|feed-consumer
+    system-log-limit: 10000
+    #system-policy-types: 1
+    #system-log-file: "./log.json"            # file path
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml]:
```suggestion
```

File: deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    system-policy-dir: "./"
+    deprecate-old-mode: true
+    namespace-filter:
+      - "!kube-system"
+    fromsource-filter:
+      - "knoxAutoPolicy"
+
+  cluster:
+    cluster-info-from: "k8sclient"            # k8sclient|accuknox
+    #cluster-mgmt-url: "http://cluster-management-service.accuknox-dev-cluster-mgmt.svc.cluster.local/cm"
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml]:
```suggestion
```

File: deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+feed-consumer:
+  driver: "pulsar" # kafka | pulsar
+  servers:
+    - "localhost:6650"
+  topic: 
+    cilium: "persistent://accuknox/datapipeline/ciliumalertsrawflow"
+    kubearmor: "persistent://accuknox/datapipeline/kubearmoralertsrawflow"
+    # cilium: "cilium-alerts"
+    # kubearmor: "kubearmor-alerts"
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml]:
```suggestion
```

File: deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  # number-of-consumers: 1
+  # consumer-group: knoxautopolicy
+  # event-buffer-size: 50
+  # kafka:
+  #   server-address-family: v4
+  #   session-timeout: 6000 # in millisecond
+  # pulsar:
+  #   connection-timeout: 10 # in second
+  #   operation-timeout: 30 # in second
+  # -----------------------------------------------
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml]:
```suggestion
```

File: deployments/discovery-engine-chart/templates/hpa.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
+{{- if .Values.autoscaling.enabled }}
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/templates/hpa.yaml]: We don't need HPA.

File: deployments/discovery-engine-chart/templates/ingress.yaml [hunk 9]
```diff
@@ -0,0 +1,10 @@
+{{- if .Values.ingress.enabled -}}
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/templates/ingress.yaml]: We don't need Ingress.

File: deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+  system-observability: true
+  network-observability: true
+
+database:
+  driver: sqlite3
+  host: 127.0.0.1
+  port: 3306
+  user: root
+  password: password
+  dbname: accuknox
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml]:
```suggestion
```

**Review by wazir-ahmed — CHANGES_REQUESTED** [review]

File: deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+  table-system-log: system_log
+  table-system-alert: system_alert
+  table-system-policy: system_policy
+
+logging:
+  level: "INFO"
+
+# kubectl -n kube-system port-forward service/hubble-relay --address 0.0.0.0 --address :: 4245:80
+cilium-hubble:
+  url: localhost
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml]:
```suggestion
  url: hubble-relay.kube-system.svc.cluster.local
  port: 80
```

File: deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+  port: 4245
+
+kubearmor:
+  url: localhost
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml]:
```suggestion
  url: kubearmor.kube-system.svc.cluster.local

```

File: deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+  table-network-log: network_log
+  table-network-policy: network_policy
+  table-system-log: system_log
+  table-system-alert: system_alert
+  table-system-policy: system_policy
+
+logging:
+  level: "INFO"
+
+# kubectl -n kube-system port-forward service/hubble-relay --address 0.0.0.0 --address :: 4245:80
```
Comment (wazir-ahmed) [inline deployments/discovery-engine-chart/configmapfiles/discovery-engine/conf.yaml]:
```suggestion
```

**Review by wazir-ahmed — CHANGES_REQUESTED** [review]

**Review by wazir-ahmed — APPROVED** [review]
PR looks good to me. Thanks @Nagarajan0396 for the contribution.

**Pull request merged**